### PR TITLE
docs: improve session

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
Small improvements or fixes.

If we agree, I would like to remve all sample code that uses `$_SESSION` directly.

**Checklist:**
- [x] Securely signed commits
- [] Component(s) with PHPDoc blocks, only if necessary or adds value
- [] Unit testing, with >80% coverage
- [x] User guide updated
- [] Conforms to style guide

### CODE REVIEW

**Review by MGatner — CHANGES_REQUESTED** [review]
I think the UG used "userdata" throughout to indicate "non-flash data stored by the framework", because some actual `$_SESSION` items are not included:
```php
$_exclude = array_merge(['__ci_vars'], $this->getFlashKeys(), $this->getTempKeys());
```
I'm fine with making this a little more opaque but we should note somewhere what is _not_ included. 

Otherwise: I am in favor of encouraging use of the Session handler over the global variable.

File: user_guide_src/source/libraries/sessions.rst [hunk 1]
```diff
@@ -75,7 +75,7 @@ However, non-blocking requests in the context of sessions also means
 unsafe, because, modifications to session data (or session ID regeneration)
 in one request can interfere with the execution of a second, concurrent
 request. This detail was at the root of many issues and the main reason why
-CodeIgniter 4 has a completely re-written Session library.
+CodeIgniter 3 has a completely re-written Session library.
```
[the remark this answers is not in the document]
Comment (kenjis) [reply user_guide_src/source/libraries/sessions.rst:78]: No, this is historical explanation.
Ci 2.x Session did not have locking, then to solve the issues, CI3 had a completely re-written Session library.

CI4 may have a completely re-written Session library, but lack of locking is not the main reason.

**Comment by kenjis** [comment]
There are some notes (important) to describe the items that the `get()` method returns
like the following:
> The get() method WILL return flashdata or tempdata items when retrieving a single item by key. It will not return flashdata or tempdata when grabbing all data from the session, however.

And I added a note for the word "userdata".

**Review by datamweb — APPROVED** [review]

**Review by MGatner — APPROVED** [review]
Sorry, I missed those other comments. Thank you for updating. LGTM!

**Pull request merged**